pkg/generators/config.go

```diff
@@ -51,36 +49,31 @@ func DefaultNameSystem() string {
 	return "sorting_namer"
 }
 
-func Packages(context *generator.Context, arguments *args.GeneratorArgs) generator.Packages {
-	boilerplate, err := arguments.LoadGoBoilerplate()
+func GetTargets(context *generator.Context, args *args.Args) []generator.Target {
+	boilerplate, err := gengo.GoBoilerplate(args.GoHeaderFile, gengo.StdBuildTag, gengo.StdGeneratedBy)
 	if err != nil {
 		klog.Fatalf("Failed loading boilerplate: %v", err)
 	}
-	header := append([]byte(fmt.Sprintf("// +build !%s\n\n", arguments.GeneratedBuildTag)), boilerplate...)
-	header = append(header, []byte(
-		`
-// This file was autogenerated by openapi-gen. Do not edit it manually!
-
-`)...)
 
 	reportPath := "-"
-	if customArgs, ok := arguments.CustomArgs.(*generatorargs.CustomArgs); ok {
-		reportPath = customArgs.ReportFilename
+	if args.ReportFilename != "" {
+		reportPath = args.ReportFilename
 	}
 	context.FileTypes[apiViolationFileType] = apiViolationFile{
 		unmangledPath: reportPath,
 	}
 
-	return generator.Packages{
-		&generator.DefaultPackage{
-			PackageName: filepath.Base(arguments.OutputPackagePath),
-			PackagePath: arguments.OutputPackagePath,
-			HeaderText:  header,
-			GeneratorFunc: func(c *generator.Context) (generators []generator.Generator) {
+	return []generator.Target{
+		&generator.SimpleTarget{
+			PkgName:       filepath.Base(args.OutputDir),
```

Comment:
is `filepath.Base(args.OutputDir)` actually correct for finding the package name, or is it best-effort / usually-correct? if the latter, comment here to note that limitation and possibly for a follow-up improvement

would `path.Base(args.OutputPackage)` be more likely to be correct (though still possibly best-effort) - `path` instead of `filepath`, `customArgs.OutputPackage` instead of `arguments.OutputBase`?

Reply:
Correct on Pkg vs Dir.

filepath touches a few more places.